File: new/atoms/reset-password.test.mjs
import { resetPassword as __pure_resetPassword } from './reset-password.mjs';
// צילום-מקומי + עטיפת-כריכה (מנוע-הקשיחים; בדיקה לא מייבאת אטום-שכן)
const __d_reset_password_T = {
  k1: "auth/user-not-found",
  k2: "לא נמצא משתמש עם האימייל הזה",
  k3: "auth/invalid-email",
  k4: "כתובת האימייל אינה תקינה",
};
const resetPassword = (...a) => __pure_resetPassword(...a, ...Array(Math.max(0, 4 - a.length)).fill(undefined), __d_reset_password_T);
let f = 0;
const ok = (cond, msg) => { if (!cond) { console.error('✗ ' + msg); f = 1; } };
const threw = async (p, expected, msg) => {
  try { await p; ok(false, msg + ' — לא נזרקה שגיאה'); }
  catch (e) { ok(e.message === expected, msg + ' — נזרק: ' + e.message); }
};
// שקעים-מיני כמתועד בחוזה
const AUTH = { tag: 'AUTH' };
const reqAuth = () => AUTH;
const rejWith = (code) => () => Promise.reject({ code });
const heb = (e) => new Error('עברית:' + (e?.code ?? '') + '|' + (e?.message ?? ''));

// 1) הצלחה — sendReset נקרא פעם-אחת עם (auth, email); נפתר undefined
{
  const calls = [];
  const sendReset = (a, em) => { calls.push([a, em]); return Promise.resolve(); };
  const r = await resetPassword('a@b.com', reqAuth, sendReset, heb);
  ok(r === undefined, 'דוגמה 1 — לא נפתר undefined');
  ok(calls.length === 1 && calls[0][0] === AUTH && calls[0][1] === 'a@b.com',
    'דוגמה 1 — sendReset: ' + JSON.stringify(calls.map(([a, em]) => [a.tag, em])));
}
// 2) user-not-found
await threw(resetPassword('x@y.com', reqAuth, rejWith('auth/user-not-found'), heb),
  'לא נמצא משתמש עם האימייל הזה', 'דוגמה 2');
// 3) invalid-email
await threw(resetPassword('לא-אימייל', reqAuth, rejWith('auth/invalid-email'), heb),
  'כתובת האימייל אינה תקינה', 'דוגמה 3');
// 4) קוד אחר ⇒ hebrewAuthError עם השגיאה המקורית
await threw(resetPassword('a@b.com', reqAuth, rejWith('auth/too-many-requests'), heb),
  'עברית:auth/too-many-requests|', 'דוגמה 4');
// 5) דחייה בלי code ⇒ code נגזר '' ⇒ hebrewAuthError, בלי נפילה
await threw(resetPassword('a@b.com', reqAuth, () => Promise.reject(new Error('boom')), heb),
  'עברית:|boom', 'דוגמה 5');
// 6) requireAuth זורק ⇒ נתפס וממופה; sendReset לא נקרא
{
  let sent = 0;
  const boomAuth = () => { throw new Error('הענן לא אותחל — פנו למנהל המערכת'); };
  await threw(resetPassword('a@b.com', boomAuth, () => { sent++; return Promise.resolve(); }, heb),
    'עברית:|הענן לא אותחל — פנו למנהל המערכת', 'דוגמה 6');
  ok(sent === 0, 'דוגמה 6 — sendReset נקרא למרות כשל-requireAuth');
}

if (f) process.exit(1);
console.log('✓ reset-password: 6 דוגמאות-חוזה — ירוק');

File: new/dart-maor/reset-password_test.dart
import '../dart-data-maor/reset-password-sockets.dart' as sk_reset_password;
// בדיקת-חוזה (רתמת-זהב) · resetPassword — מייבאת אך ורק את האטום-שלה (חוק-4).
// 6 דוגמאות-החוזה זהות למקור-ה-JS new/atoms/reset-password.test.mjs.
// כל קריאות-הענן מוזרקות כשקעים (requireAuth · sendReset · hebrewAuthError).
// הרצה: dart run --enable-asserts new/dart-maor/reset-password_test.dart  ⇒ exit 0
import 'reset-password.dart';

// שגיאת-דחייה נושאת code, כמו שגיאת-Firebase (האטום קורא e?.code).
class _AuthErr {
  final String? code;
  _AuthErr(this.code);
}

Future<String> _throwMsg(Future<void> Function() f) async {
  try {
    await f();
    return '__no-throw__';
  } catch (e) {
    return e is StateError ? (e.message) : e.toString();
  }
}

Future<void> main() async {
  var n = 0;
  final auth = Object();

  // 1) הצלחה: sendReset נקרא פעם-אחת עם (auth, email); ההבטחה נפתרת
  final calls = <List>[];
  Future<void> sendOk(dynamic a, String e) async => calls.add([a, e]);
  await resetPassword('a@b.com', () => auth, sendOk, (e) => StateError('unmapped'), sk_reset_password.resetPassword_T);
  if (calls.length != 1 ||
      !identical(calls[0][0], auth) ||
      calls[0][1] != 'a@b.com') {
    throw StateError('FAIL ex1');
  }
  n++;

  // 2) code=user-not-found ⇒ הודעה עברית ספציפית
  final m2 = await _throwMsg(() => resetPassword('a@b.com', () => auth,
      (a, e) => Future.error(_AuthErr('auth/user-not-found')),
      (e) => StateError('unmapped'), sk_reset_password.resetPassword_T));
  if (m2 != 'לא נמצא משתמש עם האימייל הזה') throw StateError('FAIL ex2 ⇒ $m2');
  n++;

  // 3) code=invalid-email ⇒ הודעה עברית ספציפית
  final m3 = await _throwMsg(() => resetPassword('a@b.com', () => auth,
      (a, e) => Future.error(_AuthErr('auth/invalid-email')),
      (e) => StateError('unmapped'), sk_reset_password.resetPassword_T));
  if (m3 != 'כתובת האימייל אינה תקינה') throw StateError('FAIL ex3 ⇒ $m3');
  n++;

  // 4) code אחר ⇒ בדיוק תוצאת hebrewAuthError(e)
  final m4 = await _throwMsg(() => resetPassword('a@b.com', () => auth,
      (a, e) => Future.error(_AuthErr('auth/too-many-requests')),
      (e) => StateError('HEB:${(e as dynamic).code}'), sk_reset_password.resetPassword_T));
  if (m4 != 'HEB:auth/too-many-requests') throw StateError('FAIL ex4 ⇒ $m4');
  n++;

  // 5) דחייה בלי code ⇒ code נגזר '' ⇒ תוצאת hebrewAuthError (לא נפילה)
  final m5 = await _throwMsg(() => resetPassword('a@b.com', () => auth,
      (a, e) => Future.error(_AuthErr(null)), (e) => StateError('mapped-null'), sk_reset_password.resetPassword_T));
  if (m5 != 'mapped-null') throw StateError('FAIL ex5 ⇒ $m5');
  n++;

  // 6) requireAuth זורק ⇒ נתפס+ממופה; sendReset לא נקרא כלל
  var called6 = false;
  final m6 = await _throwMsg(() => resetPassword('a@b.com', () => throw _AuthErr('auth/not-init'),
      (a, e) async => called6 = true,
      (e) => StateError('mapped6'), sk_reset_password.resetPassword_T));
  if (m6 != 'mapped6' || called6) throw StateError('FAIL ex6 (called=$called6)');
  n++;

  assert(n == 6, 'assert-live guard');
  print('OK resetPassword: $n asserts passed');
}
